src/textual/widgets/_data_table.py

```diff
@@ -26,40 +29,114 @@
 from ..scroll_view import ScrollView
 from ..strip import Strip
 
+CellCacheKey: TypeAlias = "tuple[RowKey, ColumnKey, Style, bool, bool, int]"
+LineCacheKey: TypeAlias = (
+    "tuple[int, int, int, int, Coordinate, Coordinate, Style, CursorType, bool, int]"
+)
+RowCacheKey: TypeAlias = (
+    "tuple[RowKey, int, Style, Coordinate, Coordinate, CursorType, bool, bool, int]"
+)
 CursorType = Literal["cell", "row", "column", "none"]
-CELL: CursorType = "cell"
 CellType = TypeVar("CellType")
 
 
 class CellDoesNotExist(Exception):
-    pass
+    """Raised when the user supplies coordinates or cell keys which
+    do not exist in the DataTable."""
 
 
-def default_cell_formatter(obj: object) -> RenderableType | None:
-    """Format a cell in to a renderable.
+class DuplicateKey(Exception):
+    """Raised when the RowKey or ColumnKey provided already refers to
+    an existing row or column in the DataTable. Keys must be unique."""
+
+
+@functools.total_ordering
+class StringKey:
+    """An object used as a key in a mapping. It can optionally wrap a string,
+    and lookups into a map using the object behave the same as lookups using
+    the string itself."""
```

Comment:
May want reformatting to confirm with [our docstring guidelines](https://github.com/Textualize/textual/wiki/Documentation-standards#class).